jobs/build/ocp4/build.groovy

```diff
@@ -467,15 +467,22 @@ def stageSyncImages() {
         echo "No built images to sync."
         return
     }
-    if(buildPlan.dryRun) {
-        echo "Not syncing images in a dry run."
-        return
+
+    def record_log = buildlib.parse_record_log(doozerWorking)
```

Comment:
I wonder if this could be made more like scan-sources instead of relying on record logs. e.g. a verb smart enough that if any operator / operand image in the assembly is newer than the available bundle matching the NVR of the operator, it rebuilds the bundle. 
`olm-bundle:scan-and-rebuild`.

Reply:
It is possible. PR https://github.com/openshift/doozer/pull/498 made a change that if we run `doozer --group $group --assembly $name olm-bundle:rebase-and-build` without `--force` option or `NVRs`, it will check if the bundle for assembly-selected operator exists, and only rebase and build if that doesn't exist.

This part is to pre-build the bundles for operators that haven't been pinned to the assembly config. The built bundles will not be selected unless their operators are selected.